assets/blocks/button/color-props.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	getColorClassName,
+	getColorObjectByAttributeValues,
+} from '@wordpress/block-editor';
+
+/**
+ * Get className for gradient.
+ *
+ * @param {string} gradientSlug
+ * @return {string|undefined} Class.
+ */
+export const getGradientClass = ( gradientSlug ) => {
+	if ( ! gradientSlug ) {
+		return undefined;
+	}
+	return `has-${ gradientSlug }-gradient-background`;
+};
+
+/**
+ * Get className and style from color settings.
+ *
+ * Copied from Gutenberg color support hook.
+ *
+ * @param {Object} props
+ * @param {Object} props.attributes Block Attributes.
+ * @param {Object} props.colors     Theme color settings.
+ */
+export const getColorAndStyleProps = ( { attributes, colors } ) => {
```

Comment:
This is the same code running in the color support hook that processes the selected color settings stored as attributes and builds CSS classes and inline styles. We use this to do the same mapping and use the styles for the inner button element.
(Core button copies this one too, for the same reasons.)